packages/asu-core/__tests__/aws/services/cache.service.spec.ts

```diff
@@ -0,0 +1,149 @@
+        it('should call given callback when cache get returns null', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockResolvedValue(null);
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+        });
+
+        it('should call given callback when cache get throws exception', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockImplementation(() => {
+                throw 'Failed to get value';
+            });
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+        });
+
+        it('should set value if key was not previously cached', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockResolvedValue(null);
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+            expect(cacheClient.set.mock.calls[0][0]).toBe(expectedKey);
+            expect(cacheClient.set.mock.calls[0][1]).toBe(expectedValue);
+            expect(cacheClient.set.mock.calls[0][2]).toBe('EX');
+            expect(cacheClient.set.mock.calls[0][3]).toBe(TimeToLive.OneHour);
+        });
+
+        it('should set value with given TimeToLive', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockResolvedValue(null);
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            }, TimeToLive.FiveMinutes);
+
+            // Assert
+            expect(cacheClient.set.mock.calls[0][3]).toBe(TimeToLive.FiveMinutes);
+        });
+
+        it('should still return value when cache set throws exception', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockResolvedValue(null);
+            cacheClient.set.mockImplementation(() => {
+                throw 'Failed to get value';
```

Comment:
minor typo for the error being throw during `set`
```suggestion
                throw 'Failed to set value';
```

Reply:
Good catch, thanks